CHANGELOG.md

```diff
@@ -17,6 +17,7 @@ Mapbox welcomes participation and contributions from everyone.
 - Added Android 13 support. [#6196](https://github.com/mapbox/mapbox-navigation-android/pull/6196)
 - Declared [POST_NOTIFICATIONS](https://developer.android.com/reference/android/Manifest.permission#POST_NOTIFICATIONS) permission in SDK's AndroidManifest.xml. It is highly recommended for apps to request the permission in runtime. Without it, the SDK will not be able to show the notification with trip progress in the notification drawer for apps that target Android 13 or higher. [#6196](https://github.com/mapbox/mapbox-navigation-android/pull/6196)
 - Marked `MapboxNavigationApp` and `MapboxMavigationObserver` methods with `@UiThread` annotations. [#6254](https://github.com/mapbox/mapbox-navigation-android/pull/6254)
+- Fixed `RouteAlternativesObserver`: does not spread alternative if set to `MapboxNavigation#setRoutes` has been received alternatives plus one or a few additional routes. [#6214](https://github.com/mapbox/mapbox-navigation-android/pull/6214)
```

Comment:
Does anything actually change for developers? Weren't our flag (removed in this PR) prevent this from happening, or where there edge cases that we did not consider that are fixed now?

Reply:
The string is still here 🙃

Reply:
🤔 it struggles!